spec-tests/src/test/java/org/candlepin/spec/ResponseFilteringSpecTest.java

```diff
@@ -0,0 +1,174 @@
+    }
+
+    @Test
+    void shouldAllowMultipleExcludes() {
+        ConsumerDTO consumerFull = admin.consumers().getConsumer(consumer.getUuid());
+        ConsumerDTO consumerFiltered = Request.from(admin)
+            .addQueryParam("exclude", "href")
+            .addQueryParam("exclude", "facts")
+            .setPath("/consumers/" + consumer.getUuid())
+            .execute()
+            .deserialize(ConsumerDTO.class);
+
+        assertThat(consumerFull)
+            .usingRecursiveComparison()
+            .ignoringFields("href")
+            .ignoringFields("facts")
+            .isEqualTo(consumerFiltered);
+        assertThat(consumerFiltered.getHref()).isNull();
+        assertThat(consumerFiltered.getFacts()).isNull();
+    }
+
+    @Test
+    void shouldAllowSingleInclude() {
+        JsonNode consumerFiltered = Request.from(admin)
+            .addQueryParam("include", "href")
+            .setPath("/consumers/" + consumer.getUuid())
+            .execute()
+            .deserialize(JsonNode.class);
+
+        assertThat(consumerFiltered).hasSize(1);
+        assertThat(consumerFiltered.get("href")).isNotNull();
+    }
+
+    @Test
+    void shouldAllowMultipleIncludes() {
+        JsonNode consumerFiltered = Request.from(admin)
+            .addQueryParam("include", "href")
+            .addQueryParam("include", "facts")
+            .setPath("/consumers/" + consumer.getUuid())
+            .execute()
+            .deserialize(JsonNode.class);
+
+        assertThat(consumerFiltered).hasSize(2);
+        assertThat(consumerFiltered.get("href")).isNotNull();
+        assertThat(consumerFiltered.get("facts")).isNotNull();
+    }
+
+    @Test
+    void shouldAllowFiltersOnEncapsulatedObjects() {
+        JsonNode consumerFiltered = Request.from(admin)
+            .addQueryParam("include", "id")
+            .addQueryParam("include", "owner.id")
+            .setPath("/consumers/" + consumer.getUuid())
+            .execute()
+            .deserialize(JsonNode.class);
+
+        assertThat(consumerFiltered).hasSize(2);
+        assertThat(consumerFiltered.get("id")).isNotNull();
+        assertThat(consumerFiltered.get("owner")).isNotNull();
+        assertThat(consumerFiltered.get("owner")).hasSize(1);
+        assertThat(consumerFiltered.get("owner").get("id")).isNotNull();
+    }
+
+    @Test
+    void shouldAllowFiltersOnEncapsulatedLists() {
+        admin.consumers().createConsumer(Consumers.random(owner));
+        admin.consumers().createConsumer(Consumers.random(owner));
+        JsonNode consumers = Request.from(admin)
+            .addQueryParam("type", "system")
+            .addQueryParam("include", "id")
+            .addQueryParam("include", "owner.id")
+            .setPath("/consumers")
+            .execute()
+            .deserialize(JsonNode.class);
+
+        assertThat(consumers)
+            .isNotEmpty()
+            .allSatisfy(consumer -> {
+                assertThat(consumer.get("id")).isNotNull();
```

Comment:
This should also be asserting that each consumer object only has 2 fields

Reply:
Done.